test/suites/profiling.bash

```diff
@@ -6,6 +6,12 @@ SUITE_profiling_PROBE() {
     if $COMPILER_TYPE_CLANG && ! command -v llvm-profdata$CLANG_VERSION_SUFFIX >/dev/null; then
         echo "llvm-profdata$CLANG_VERSION_SUFFIX tool not found"
     fi
+
+    beginswith() { case $2 in "$1"*) true;; *) false;; esac; }
+    ver=$(echo $compiler_version | sed 's/.*clang version //')
```

Comment:
This may work for open source clang, but will not identify all versions of Apple clang. The versions of Apple clang that shipped with Xcode 4.6 through 10.3 inclusive produce a version string that identifies it as "Apple LLVM" rather than "clang". Apple clang in Xcode 3.2.3 through 4.5.2 inclusive and in Xcode 11 and later calls itself "Apple clang". Clang in Xcode 3.2, 3.2.1, and 3.2.2 calls itself just "clang". You can see the Apple clang version strings here: https://trac.macports.org/wiki/XcodeVersionInfo

Note that Apple clang uses a different version numbering scheme than open source clang.

Reply:
Yes you are totally right. And I don't particularly like this fix anyway. But it works for linux and it doesn't break anything on apple clang since everything based on clang 4+ is called 7+ on apple. (It doesn't fix apple clang either, but one step at a time...).